apps/aecore/lib/aecore/sync/sync.ex

```diff
@@ -0,0 +1,756 @@
+    pick_from = Enum.filter(pool, fn {_, _, elem} -> elem == false end)
+
+    random =
+      pick_from
+      |> length()
+      |> :rand.uniform()
+
+    {pick_height, pick_hash, false} = Enum.fetch!(pick_from, random - 1)
+    Logger.info("#{__MODULE__}: Get block at height #{pick_height}")
+    {{:get_block, pick_height, pick_hash}, st}
+  end
+
+  def get_next_work_item(%Task{} = st) do
+    Logger.info("#{__MODULE__}: Take a break, nothing to do: #{inspect(st)}")
+    {:take_a_break, st}
+  end
+
+  def do_handle_worker({:new_worker, peer_id, pid}, %Task{workers: ws} = st) do
+    case Enum.filter(ws, fn {p_id, _} -> p_id == peer_id end) do
+      [] ->
+        :ok
+
+      [{_, old}] ->
+        Logger.info(
+          "#{__MODULE__}: Peer: #{inspect(peer_id)} already has a worker: #{inspect(old)}"
+        )
+    end
+
+    Process.link(pid)
+    %Task{st | workers: Task.keystore(peer_id, {peer_id, pid}, ws)}
+  end
+
+  def do_handle_worker({:change_worker, peer_id, old_pid, new_pid}, %Task{workers: ws} = st) do
+    case Enum.filter(ws, fn {p_id, _} -> p_id == peer_id end) do
+      [] ->
+        Logger.info("#{__MODULE__}: Missing worker #{old_pid} for peer #{inspect(peer_id)}")
+
+      [{_, ^old_pid}] ->
+        :ok
+
+      [{_, another_pid}] ->
+        Logger.info(
+          "#{__MODULE__}: Wrong worker stored for peer #{inspect(peer_id)} (#{another_pid})"
+        )
+    end
+
+    Process.link(new_pid)
+    # The old worker will terminate right after the call to handle_worker is finished
+    # so we can safely remove it
+    Process.unlink(old_pid)
+
+    %Task{st | workers: Task.keystore(peer_id, {peer_id, new_pid}, ws)}
+  end
+
+  @spec do_terminate_worker(pid(), Sync.t()) :: Sync.t()
+  def do_terminate_worker(pid, %Sync{sync_tasks: sts} = state) do
+    case Enum.filter(sts, fn %{workers: {_, p}} -> p == pid end) do
+      [st] ->
+        pid
+        |> do_terminate_worker(st)
+        |> Task.set_sync_task(state)
+
+      [] ->
+        state
+    end
+  end
+
+  def do_terminate_worker(pid, %Task{workers: ws} = st) do
+    [{peer, _}] = Enum.filter(ws, fn {_, p} -> p == pid end)
+    Logger.info("#{__MODULE__}: Terminating worker: #{pid} for worker: #{peer}")
+    %Task{st | workers: Enum.filter(ws, fn {_, p} -> p != pid end)}
+  end
+
+  @spec ping_peer(peer_id()) :: :ok | {:error, String.t()}
+  def ping_peer(peer_id) do
+    res = PeerConnection.ping(peer_id)
+
+    case res do
+      :ok ->
+        Logger.info("#{__MODULE__}: Pinged peer #{inspect(peer_id)} successfully")
+        :ok
+
+      ## Log ping
+      {:error, reason} = err ->
+        Logger.info("#{__MODULE__}: Error while pinging peer #{inspect(peer_id)}: #{reason}")
+        err
+    end
+  end
+
+  def do_forward_block(block, peer_id) do
+    PeerConnection.send_new_block(block, peer_id)
+  end
+
+  def do_forward_tx(tx, peer_id) do
+    PeerConnection.send_new_tx(peer_id, tx)
```

Comment:
Does this function exist `PeerConnection.send_new_tx`? I tried to test it running separate nodes and the same error appears:
https://github.com/aeternity/elixir-node/issues/560

Reply:
Yes, now we saw that the function is no longer available since PR #557. We will fix this now.

Reply:
And one little remark - the arguments are not in the right order, should be PeerConnection.send_new_tx(tx, peer_id)